public_html/wp-content/themes/wporg-events-2023/postcss/page/front-page/cover.pcss

```diff
@@ -1,68 +1,73 @@
 .wporg-events__cover {
 
-	.wp-block-columns .wp-block-column:first-child {
+	@media (--large) {
 		padding-left: var(--wp--preset--spacing--edge-space);
 		padding-right: var(--wp--preset--spacing--edge-space);
 	}
 
-	@media (--medium) {
-		.wp-block-columns {
-			flex-wrap: wrap !important;
+	.wp-block-columns {
 
-			.wp-block-column {
-				flex-grow: 1 !important;
-			}
+		@media (--medium) {
+			flex-wrap: wrap !important;
 		}
-	}
 
-	@media (--large) {
-		padding-left: var(--wp--preset--spacing--edge-space);
-		padding-right: var(--wp--preset--spacing--edge-space);
-
-		.wp-block-columns {
+		@media (--large) {
 			flex-wrap: nowrap !important;
 		}
 
-		.wp-block-columns .wp-block-column:first-child {
-			padding-left: 0;
-			padding-right: 0;
-		}
+		.wp-block-column {
+			&:first-child {
+				padding-left: var(--wp--preset--spacing--edge-space);
+				padding-right: var(--wp--preset--spacing--edge-space);
 
-		.wp-block-columns .wp-block-column:first-child > p {
-			max-width: 460px;
-		}
-	}
+				@media (--large) {
+					padding-left: 0;
+					padding-right: 0;
 
-	.wp-block-heading em {
-		display: block;
-	}
+					> p {
+						max-width: 460px;
+					}
+				}
 
-	@media (--huge) {
-		.wp-block-columns .wp-block-column:first-child {
-			flex-basis: 33% !important;
-		}
+				@media (--huge) {
+					flex-basis: 33% !important;
+				}
+			}
+
+			&:last-child {
+
+				@media (--huge) {
+					flex-basis: 66.665% !important;
+				}
+			}
 
-		.wp-block-columns .wp-block-column:last-child {
-			flex-basis: 66.665% !important;
+			@media (--medium) {
+				flex-grow: 1 !important;
+			}
+
+			.wp-block-heading em {
+				display: block;
+			}
 		}
 	}
 
-	& .wp-block-wporg-google-map {
```

Comment:
🤔 I thought PostCSS required using `&`. Is that no longer the case?

https://preset-env.cssdb.org/features/#nesting-rules

Reply:
I remember for PostCSS you can use & after installing plugins like postcss-nested or postcss-scss, and it seems there hasn't been a rule that & is mandatory except for pseudo-classes.

And when looking into this, though I still couldn't find any clear statements stating that & must or must not be used, I discovered that native CSS has started supporting the use of nesting directly. It's explicitly mentioned [here](https://developer.chrome.com/docs/css-ui/css-nesting#nesting_examples) that using & and not using it is equivalent. Major browsers have already begun [supporting](https://caniuse.com/css-nesting) this feature.

Also I noticed that in the link you shared, there's a "specification" linking to the same [W3C draft](https://www.w3.org/TR/css-nesting-1/) that the Google Docs references. But I also didn't find any clear instructions in that draft saying that the & is absolutely required or not required, though.

Reply:
It looks like browser support for it without `&` is only 2%, so it seems like we won't be able to use that for a long time. Can you check the built CSS to see if it gets converted to unnested rules when `&` is not used?

If it is being converted to unnested vanilla CSS, then removing the `&` seems fine, but otherwise I think we should keep it so that the layout doesn't break for folks using recent browsers.

Reply:
We should try to remove nesting where possible. Additionally, I would vote to update this theme to match our other themes which use .scss. This potential "gotcha" for future developers is no longer an issue and we have more consistency across the child themes.

Reply:
> It looks like browser support for it without & is only 2%, so it seems like we won't be able to use that for a long time. Can you check the built CSS to see if it gets converted to unnested rules when & is not used?

I mentioned the support for nesting CSS earlier just to share that eventually, we will be able to write nested CSS directly in native CSS. But I think that before W3C officially transitions from a Working Draft to a Standard, these Preprocessor Tools (like SCSS, PostCSS) won't convert styles into nested CSS. And I feel that in the future, they might not even need to, unless converting to nested CSS proves to be more efficient. As it stands, the current method of conversion (completely unnested CSS) is already supported by all browsers.

I took a look at the built CSS here, and the currently converted vanilla CSS is still unnested.

Reply:
> We should try to remove nesting where possible.

Why is that? I think it's a convenient way to keep things properly scoped, and it's makes things much easier to read for me, similar to indenting a `for` loop in PHP.


> update this theme to match our other themes which use .scss.

What's gotcha are you referring to? 

I think the ideal would be the opposite: [migrating SASS themes to PostCSS](https://github.com/WordPress/wporg-parent-2021/issues/1). 

We originally agreed on doing that when we built the News theme, but then ended up using SASS b/c it was copy/pasta from Blockbase, and then we were too far into it make the switch while we had deadlines.

In my mind [SASS is tech debt while PostCSS is future-proofed](https://github.com/WordPress/gutenberg/issues/14801#issuecomment-480087247).

Reply:
> I think the ideal would be the opposite: [migrating SASS themes to PostCSS](https://github.com/WordPress/wporg-parent-2021/issues/1).
> 
> We originally agreed on doing that when we built the News theme, but then ended up using SASS b/c it was copy/pasta from Blockbase, and then we were too far into it make the switch while we had deadlines.
> 
> In my mind [SASS is tech debt while PostCSS is future-proofed](https://github.com/WordPress/gutenberg/issues/14801#issuecomment-480087247).

TIL, thanks for these links

Reply:
> We should try to remove nesting where possible.
> Why is that? I think it's a convenient way to keep things properly scoped, and it's makes things much easier to read for me, similar to indenting a for loop 

- Nesting typically leads to longer than necessary css chains
- Nesting typically reduces the ability reuse css, leading to more styles
  - We try to stick with BEM, [Gutenberg's chosen naming convention](https://developer.wordpress.org/block-editor/contributors/code/coding-guidelines/#css)
- Nested CSS can be harder to read

> What's gotcha are you referring to?

This conversation about whether & is needed for nested selectors.

> I think the ideal would be the opposite: https://github.com/WordPress/wporg-parent-2021/issues/1.

I'm not suggesting we need to stick with .scss. I'm suggesting we remain consistent. As Gutenberg matures, I expect we will be writing less CSS and feel consistency (provided it is working and not an obvious dead end) trumps difference. If we want to transition away from .scss, let's log that so we can all be on the same page.